apm-agent-plugins/apm-es-restclient-plugin/apm-es-restclient-plugin-5_6/src/main/java/co/elastic/apm/agent/es/restclient/v5_6/ElasticsearchClientAsyncInstrumentation.java

```diff
@@ -79,6 +79,7 @@ public ElasticsearchClientAsyncInstrumentation(ElasticApmTracer tracer) {
     public static class ElasticsearchRestClientAsyncAdvice {
         @VisibleForAdvice
         public static final GlobalThreadLocal<Span> spanTls = GlobalThreadLocal.get(ElasticsearchRestClientAsyncAdvice.class, "spanTls");
+
         @AssignTo.Argument(5)
         @Advice.OnMethodEnter(suppress = Throwable.class)
         public static ResponseListener onBeforeExecute(@Advice.Argument(0) String method,
```

Comment:
[minor] The best practice would be to return an `Object[]` instead of using `GlobalThreadLocal`

Reply:
How would that look like? We would use `Object[]` as the `@Advice.Enter` type in the exit advice?
Let's not do that now, it means either adding allocation for the Object array or complication of reusing such (probably through another `GlobalThreadLocal` 🙂  )

Reply:
> How would that look like? We would use Object[] as the @Advice.Enter type in the exit advice?

Exactly.

> adding allocation for the Object array

Adding something to a Global/DetachedThreadLocal also allocates a `WeakKey` when calling `WeakConcurrentMap#put`. The JIT should be able to easily replace the `Object[]` allocation with stack allocations once it inlines the indy-dispatched advices.

Reply:
> Adding something to a Global/DetachedThreadLocal also allocates a WeakKey when calling `WeakConcurrentMap#put`.

That's right, you only added key caching for lookups!

> The JIT should be able to easily replace the Object[] allocation with stack allocations once it inlines the indy-dispatched advices.

OK, as long as we don't measure through microbenchmarks without sufficient warmup 😜 

I will change that then